src/world/GameWorldView.cpp

```diff
@@ -308,7 +308,21 @@ void GameWorldView::DrawGUI(const RoadBuildState& rb, const TerrainRenderer& ter
                         default: id = 60; break;
                     }
 
-                    LOADER.GetMapImageN(id)->DrawFull(curPos);
+                    // If item at direction is a flag and not the start, draw the direction glyph halfway out
+                    if (GetWorld().GetNO(curPt)->GetType() == NOP_FLAG && curPt != rb.start)
+                    {
+                        // Last clicked position (coordinate) in the road route
+                        Point<int> lastPos = GetWorld().GetNodePos(rb.point) - offset + curOffset;
+                        Point<int> halfwayPos;
+                        halfwayPos.x = (curPos.x + lastPos.x) / 2;
```

Comment:
Easier: Replace `Point<int>` by `DrawPoint` and use `halfwayPos = (curPos + lastPos) / 2`